### DISCUSSION TITLE
Run S script automatically

### DISCUSSION THREAD

**Discussion by rafiff23** [teamalgoritma/community#764] [Q&A] [answered]
Halo, disini saya mau nanya apakah bisa kita menjalankan r script / python dengan otomatis? saya sudah consider tentang taskscheduler yang ada di R atau yg lainnya tp apakah itu bisa tetap berjalan saat kondisi laptop mati atau R session mati?. Goals dari R / python script saya disini adalah untuk mengcleansing data saya yang saya tarik dari spreadsheet kemudian diolah di R baru di rewrite di spreadsheet lagi. Apkah ada cara untuk bisa melakukan itu?

**Comment by kinantytasyao** [comment, marked as answer]
Dear @rafiff23 

Betul sekali, dengan menggunakan `taskscheduleR` kita dapat menjalankan R script secara otomatis setiap periode waktu yang telah ditentukan dan ini bisa dilakukan ketika laptop atau R session mati. Goals yang Anda sebutkan juga dapat diselesaikan di R dengan menggunakan package `googlesheets4` yaitu bisa membantu Anda membaca data dari googlesheet menggunakan fungsi `read_sheet()` lalu setelah selesai melakukan data cleansing di R Anda dapat rewrite datanya kembali ke google sheet menggunakan fungsi `sheet_write()`. Lebih lanjut terkait `googlesheets4` dapat dibaca di https://www.r-bloggers.com/2021/12/how-to-connect-r-to-google-sheets-using-googlesheets4/

Demikian yang dapat kami sampaikan. Semoga membantu!
